experimental/devices/inky/inky.go

```diff
@@ -0,0 +1,291 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package inky
+
+import (
+	"encoding/binary"
+	"fmt"
+	"image"
+	"image/color"
+	"log"
+	"time"
+
+	"periph.io/x/periph/conn"
+	"periph.io/x/periph/conn/display"
+	"periph.io/x/periph/conn/gpio"
+	"periph.io/x/periph/conn/physic"
+	"periph.io/x/periph/conn/spi"
+)
+
+const (
+	// Constants for an Inky pHAT
+	cols = 104
+	rows = 212
+
+	speed      = 488 * physic.KiloHertz
+	spiBits    = 8
+	chunkSize  = 4096
+	spiCommand = gpio.Low
+	spiData    = gpio.High
+)
+
+type Color int
+
+const (
+	Black  = iota
+	Red    = iota
+	Yellow = iota
+	White  = iota
+)
+
+var borderColor = map[Color]byte{
+	Black:  0x00,
+	Red:    0x33,
+	Yellow: 0x33,
+	White:  0xff,
+}
+
+// NewpHAT opens a handle to an Inky pHAT.
+func NewpHAT(p spi.Port, dc gpio.PinOut, reset gpio.PinOut, busy gpio.PinIn, color Color) (*Dev, error) {
```

Comment:
Since it's very much not configurable, I wonder if the pins should be specified as arguments or not. I don't know. Normally devices are not using the registry but I can make an exception here since this device is really meant to be used with SPI0 and specific GPIOs.

Reply:
Yeah, I couldn't decide here either. It would technically be possible to hook it up to different pins if you really wanted to but I'm not sure anyone ever would?

Reply:
Let's say we keep it this way, what I could recommend it to add an `example_test.go` (see other packages for example) which could look like:

```
// Make sure periph is initialized.
if _, err := host.Init(); err != nil {
  log.Fatal(err)
}
s, err := spireg.Open("SPI0.0")
if err != nil {
  log.Fatal(err)
}
dc := gpioreg.ByName("123")
if dc == nil {
  log.Fatal("expecting to find DC pin")
}
(same for reset, busy)
h, err := inky.NewpHAT(s, dc, reset, busy, Yellow)
if err != nil {
  log.Fatal(err)
}
(draw something simple)
```